src/ADC.c

```diff
@@ -0,0 +1,70 @@
+#include "stm32f0xx.h"
+#include "ADC.h"
+
+extern uint16_t ADC_Buffer[BUFFER_SIZE];
+
+static void DeInitBuffer (uint16_t *Buff){
+	for(uint32_t i = 0; i<BUFFER_SIZE;i++){
+			Buff[i] = 0;
+		}
+}
+
+static void ADC_Calibration (){
+	ADC1->CR &= ~ADC_CR_ADEN;
+	ADC1->CR |= ADC_CR_ADCAL;//WHY WONT YOU TURN TO 1
+	while((ADC1->CR & ADC_CR_ADCAL) != 0){
+		//TIME-OUT MANAGEMENT
+	}
+}
+
+static void init_GPIO (){
+	//Enable peripheral clock for GPIOA
+	RCC_AHBPeriphClockCmd(RCC_AHBPeriph_GPIOA, ENABLE);
+
+	GPIO_InitTypeDef GPIO_InitStructure;
+	GPIO_StructInit(&GPIO_InitStructure);
+
+	GPIO_InitStructure.GPIO_Pin = (GPIO_Pin_0 | GPIO_Pin_1 | GPIO_Pin_2 | GPIO_Pin_3 | GPIO_Pin_4 | GPIO_Pin_5 | GPIO_Pin_6| GPIO_Pin_7);
+	GPIO_InitStructure.GPIO_Mode = GPIO_Mode_AN;
+	GPIO_InitStructure.GPIO_Speed = GPIO_Speed_50MHz;
+	GPIO_InitStructure.GPIO_OType = GPIO_OType_PP;
+	GPIO_InitStructure.GPIO_PuPd = GPIO_PuPd_NOPULL;
+	GPIO_Init(GPIOA, &GPIO_InitStructure);
+
+}
+//ADC1 Continuous Conversion Software Trigger
+static void init_ADC (){
+	RCC->APB2ENR |= RCC_APB2RSTR_ADC1RST;
+	ADC_Calibration();
+	ADC1->CR |= ADC_CR_ADEN;
+	ADC1->CFGR1 |= ADC_CFGR1_CONT;
+	ADC1->CHSELR |= ADC_CHSELR_CHSEL7	| ADC_CHSELR_CHSEL6
+				 | ADC_CHSELR_CHSEL5 	| ADC_CHSELR_CHSEL4
+				 | ADC_CHSELR_CHSEL3 	| ADC_CHSELR_CHSEL2
+				 | ADC_CHSELR_CHSEL1 	| ADC_CHSELR_CHSEL0;
+	ADC1->IER |= ADC_IER_EOCIE | ADC_IER_EOSEQIE | ADC_IER_OVRIE;
```

Comment:
We're not using interrupts for the ADC, so these bits probably don't need to be set? Maybe see if it still works without them set, but if it doesn't, then we can just leave it.